models/stdp_symmetric_connection.h

```diff
@@ -0,0 +1,300 @@
+  Author: Ankur Sinha
+  SeeAlso: synapsedict
+*/
+
+#include <cmath>
+
+#include "connection.h"
+
+//#include "network.h"
+//#include "dictdatum.h"
+//#include "connector_model.h"
+//#include "common_synapse_properties.h"
+//#include "event.h"
+
+
+namespace nest
+{
+
+// connections are templates of target identifier type (used for pointer / target index addressing)
+// derived from generic connection template
+template < typename targetidentifierT >
+class STDPSymmetricConnection : public Connection< targetidentifierT >
+{
+
+public:
+  typedef CommonSynapseProperties CommonPropertiesType;
+  typedef Connection< targetidentifierT > ConnectionBase;
+
+  /**
+   * Default Constructor.
+   * Sets default values for all parameters. Needed by GenericConnectorModel.
+   */
+  STDPSymmetricConnection();
+
+
+  /**
+   * Copy constructor.
+   * Needs to be defined properly in order for GenericConnector to work.
+   */
+  STDPSymmetricConnection( const STDPSymmetricConnection& );
+
+  // Explicitly declare all methods inherited from the dependent base ConnectionBase.
+  // This avoids explicit name prefixes in all places these functions are used.
+  // Since ConnectionBase depends on the template parameter, they are not automatically
+  // found in the base class.
+  using ConnectionBase::get_delay_steps;
+  using ConnectionBase::get_delay;
+  using ConnectionBase::get_rport;
+  using ConnectionBase::get_target;
+
+  /**
+   * Get all properties of this connection and put them into a dictionary.
+   */
+  void get_status( DictionaryDatum& d ) const;
+
+  /**
+   * Set properties of this connection from the values given in dictionary.
+   */
+  void set_status( const DictionaryDatum& d, ConnectorModel& cm );
+
+  /**
+   * Send an event to the receiver of this connection.
+   * \param e The event to send
+   * \param t_lastspike Point in time of last spike sent.
+   * \param cp common properties of all synapses (empty).
+   */
+  void send( Event& e, thread t, double_t t_lastspike, const CommonSynapseProperties& cp );
+
+
+  class ConnTestDummyNode : public ConnTestDummyNodeBase
+  {
+  public:
+    // Ensure proper overriding of overloaded virtual functions.
+    // Return values from functions are ignored.
+    using ConnTestDummyNodeBase::handles_test_event;
+    port
+    handles_test_event( SpikeEvent&, rport )
+    {
+      return invalid_port_;
+    }
+  };
+
+  void
+  check_connection( Node& s,
+    Node& t,
+    rport receptor_type,
+    double_t t_lastspike,
+    const CommonPropertiesType& )
+  {
+    ConnTestDummyNode dummy_target;
+
+    ConnectionBase::check_connection_( dummy_target, s, t, receptor_type );
+
+    t.register_stdp_connection( t_lastspike - get_delay() );
+  }
+
+  void
+  set_weight( double_t w )
+  {
+    weight_ = w;
+  }
+
+private:
+  double_t
+  facilitate_( double_t w, double_t kplus )
+  {
+    double_t norm_w = ( w / Wmax_ ) + ( lambda_ * eta_ * kplus );
+    return norm_w < 1.0 ? norm_w * Wmax_ : Wmax_;
```

Comment:
I believe there is a small error here, caused by the weight maximization. Let me demonstrate here, but the same holds for depression. 

For given values of w  and kplus, you are assigning the new w to be (assuming you don't hit the ``norm_w < 1.0``):

```
w_returned 
= [( w_old / Wmax_ ) + ( lambda_ * eta_ * kplus )] * Wmax_ 
= w_old + lambda_ * eta_ * kplus * Wmax
```
This makes the weight updates scale additionally with Wmax, which I believe should not be the case. For facilitation this should rather be: 
```
w_returned 
= w_old + lambda_ * eta_ * kplus 
```

However, looking at the [corresponding line in stdp_connection](https://github.com/nest/nest-simulator/blob/master/models/stdp_connection.h#L179) for the additive case (``mu_plus=0``) I think we get the same scaling with Wmax. @abigailm @mhelias am I getting something wrong or is this an intended behavior?